src/box/xrow_update_field.c

```diff
@@ -244,7 +245,15 @@ xrow_update_mp_read_scalar(const char **expr,
 		memset(&str->array[1], 0, sizeof(str->array[1]) * 2);
 		ret->str.type = XROW_UPDATE_STRING_SINGLE;
 		break;
-		}
+	}
+	case MP_BIN: {
+		ret->type = XUPDATE_TYPE_BIN;
+		struct xrow_update_string *str = &ret->str;
+		str->array[0].data = mp_decode_bin(expr, &str->array[0].length);
+		memset(&str->array[1], 0, sizeof(str->array[1]) * 2);
+		ret->str.type = XROW_UPDATE_STRING_SINGLE;
+		break;
+	}
```

Comment:
I guess we can reduce code duplication. For example here we can introduce `xrow_update_mp_read_str` which will call `mp_decode_bin` or ` mp_decode_str` depending on type.

Similarly I would not indroduce `xrow_update_mp_read_bin` and `xrow_update_store_varbinary`. They are too similar to string ones. Basically the only difference is in encoding/decoding length.